core/src/main/java/org/apache/calcite/sql/fun/SqlStdOperatorTable.java

```diff
@@ -482,6 +500,7 @@
    * <p>Its precedence is less than the prefix {@link #UNARY_PLUS +}
    * and {@link #UNARY_MINUS -} operators.
    */
+  @LibraryOperator(libraries = {STANDARD})
   public static final SqlBinaryOperator MINUS =
       new SqlMonotonicBinaryOperator(
           "-",
```

Comment:
So does it mean that not all operators in SqlStdOperatirTable belong to STANDARD?

Reply:
Right, as @julianhyde pointed out, some operators are only for internal use (https://issues.apache.org/jira/browse/CALCITE-4363?focusedCommentId=17223122&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-17223122).
In addition, some operators are supported by most databases, but not all databases (e.g. sqrt).

Reply:
Let's not all all these annotations.

Reply:
I don't think we complicates the core data structure just in order to add a tool.

Reply:
@julianhyde Sure. We should only annotate operators that are truly 'standard'.

Reply:
@danny0405 The essential change to the 'core data structure' is just annotating some operators as 'standard'. This simply attaches richer information to the operators, which will help users.

Reply:
We should not annotate anything, the class name `SqlStdOperatorTable` already indicates that the operators are recognized as a std operators for Calcite core, if it is not, we should not put the operator in this class.

If you just want an operator table, build a static map of whatever you like in you own project, there is no need to modify this class like this.

-1 for this change.

Reply:
> We should not annotate anything, 
I am sorry I do not understand why. Could you please explain this? @danny0405

Reply:
Please move this discussion to JIRA. It is design, not code review. It needs to exist after this bug is fixed.